Source/TextKit/ASTextKitContext.mm

```diff
@@ -34,14 +34,17 @@ - (instancetype)initWithAttributedString:(NSAttributedString *)attributedString
 
 {
   if (self = [super init]) {
+    static AS::Mutex *mutex = NULL;
     static dispatch_once_t onceToken;
-    static AS::Mutex *mutex;
+    // Concurrently initialising TextKit components crashes (rdar://18448377) so we use a global lock.
     dispatch_once(&onceToken, ^{
-      mutex = new AS::Mutex();
+      if (!ASActivateExperimentalFeature(ASExperimentalRemoveTextKitInitialisingLock)) {
+        mutex = new AS::Mutex();
+      }
     });
-    
-    // Concurrently initialising TextKit components crashes (rdar://18448377) so we use a global lock.
-    AS::MutexLocker l(*mutex);
+    if (mutex) {
```

Comment:
Nit: Let's use `mutex != NULL` to be consistent with the check at the end of this scope.